src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeAssemblyBuilder.cs

```diff
@@ -270,6 +270,11 @@ internal void CheckTypeNameConflict(string strTypeName, TypeBuilder? enclosingTy
         /// <returns>Dynamic module with the specified name.</returns>
         protected override ModuleBuilder? GetDynamicModuleCore(string name)
         {
+            if (name[0] == '\0')
+            {
+                throw new ArgumentException(SR.Argument_InvalidName, nameof(name));
+            }
+
```

Comment:
```suggestion
```
Adding this validation to the runtime builder would a technical breaking change. This works fine in .NET 7, but it would start throwing exception after this change:

```
using System.Reflection;
using System.Reflection.Emit;

AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("xxx"), AssemblyBuilderAccess.Run);
ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule("MyDynamicModule");

Console.WriteLine(assemblyBuilder.GetDynamicModule("\0"));
```
I think we should avoid any observable behavior changes in runtime builders at this point.